File: local_examples/client-specific/nodejs/landing.js
// EXAMPLE: landing
// BINDER_ID nodejs-landing
// STEP_START connect
import { createClient } from 'redis';

const client = createClient();

client.on('error', err => console.log('Redis Client Error', err));

await client.connect();
// STEP_END

// STEP_START set_get_string
await client.set('key', 'value');
const value = await client.get('key');
console.log(value); // >>> value
// STEP_END

// STEP_START set_get_hash
await client.hSet('user-session:123', {
    name: 'John',
    surname: 'Smith',
    company: 'Redis',
    age: 29
})

let userSession = await client.hGetAll('user-session:123');
console.log(JSON.stringify(userSession, null, 2));
/* >>>
{
  "surname": "Smith",
  "name": "John",
  "company": "Redis",
  "age": "29"
}
 */
// STEP_END

// STEP_START close
await client.quit();
// STEP_END

File: local_examples/client-specific/ruby/landing.rb
# EXAMPLE: landing
# BINDER_ID ruby-landing
# KERNEL_NAME ruby3
# STEP_START connect
require 'redis'
r = Redis.new
# STEP_END

# STEP_START set_get_string
r.set 'foo', 'bar'
value = r.get('foo')
puts value
# STEP_END

# STEP_START hash_operations
r.hset 'user-session:123', 'name', 'John'
r.hset 'user-session:123', 'surname', 'Smith'
r.hset 'user-session:123', 'company', 'Redis'
r.hset 'user-session:123', 'age', 29

hash_value = r.hgetall('user-session:123')
puts hash_value
# STEP_END

# STEP_START close
r.close()
# STEP_END
